### PR TITLE
fix: ignore header extradata validation in goerli pre-merge

### LINKED ISSUE #2176: Make extradata verification optional
### Describe the feature

This is not always true:

https://github.com/paradigmxyz/reth/blob/2df16021bfdfaf6565662dc8afece6a5f2143ff7/crates/consensus/common/src/validation.rs#L36-L40

on goerli for example extradata can be larger (clique consensus)

this is only enforced in beacon + ethash consensus 

https://github.com/ethereum/go-ethereum/blob/2adce0b06640aa665706d014a92cd06f0720dcab/consensus/clique/clique.go#L246-L246

but not in clique:

https://github.com/ethereum/go-ethereum/blob/2adce0b06640aa665706d014a92cd06f0720dcab/consensus/ethash/consensus.go#L270-L272

## TODO 

make this configurable, perhaps via ChainSpec pre merge?

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Rjected**
We'd like to support goerli, so this seems like the best route to doing so without implementing clique. This extracts header extradata validation into a method which is only called if the network is not pre-merge goerli.

This field is now checked in the `Consensus` implementation for `BeaconConsensus`, instead of in `validate_header_standalone`.

Fixes #2176

### CODE REVIEW

**Comment by Rjected** [comment]
@gakonst Tried to sync up to block 100k, headers stage worked! But execution did not:
```
cargo run -- node --chain goerli --debug.tip 0x94b9c7be22783a3ee1e1f1f31e35f7de4612905d6fd24d3fe90a26091dca43fe
```
```
2023-04-11T02:42:00.009369Z  INFO Executing stage stage=Execution
2023-04-11T02:42:00.009385Z  INFO Stage committed progress stage=Execution checkpoint=5000
2023-04-11T02:42:00.009401Z  INFO Stage committed progress stage=Execution checkpoint=10000
Error: A stage encountered an irrecoverable error.

Caused by:
   0: Stage encountered a execution error in block 12841: Block gas used 3198856 is different from expected gas used 3231496..
   1: Block gas used 3198856 is different from expected gas used 3231496.

Location:
    bin/reth/src/node/mod.rs:346:9
```

**Comment by gakonst** [comment]
That seems to be the [first non-empty block](https://goerli.etherscan.io/block/12841) on Goerli, with [this](https://goerli.etherscan.io/tx/0x54ad70894edf7a04c591cd4e0f36d3a5f6e91f7c686e37e6f91161668ec120c6#statechange) transaction:

```
cast run 0x54ad70894edf7a04c591cd4e0f36d3a5f6e91f7c686e37e6f91161668ec120c6 -v                                    
Executing previous transactions from the block.
Traces:
  [2441700] → new <Unknown>@0xC0FfeeaCBf2611e748F08294D90bcEa6c6E16ad1
    └─ ← 11198 bytes of code


Transaction successfully executed.
Gas used: 2673956
```

Can you investigate in case this is some low hanging fruit? Else @rakita can investigate after but don't want to distract him from the current mainnet critical path optimizations right now.

**Comment by Rjected** [comment]
> Can you investigate in case this is some low hanging fruit? Else @rakita can investigate after but don't want to distract him from the current mainnet critical path optimizations right now. This is a good candidate for our `--debug.hook <transaction>` `--debug.print-inspector` to check if this touches the right storage slot with the right value?

Yeah that sounds good, will take a look / try out the flags

File: crates/consensus/beacon/src/beacon_consensus.rs [hunk 1]
```diff
@@ -53,12 +55,22 @@ impl Consensus for BeaconConsensus {
...
             // https://eips.ethereum.org/EIPS/eip-4399#using-mixhash-field-instead-of-difficulty
         } else {
             // TODO Consensus checks for old blocks:
             //  * difficulty, mix_hash & nonce aka PoW stuff
             // low priority as syncing is done in reverse order
+
+            // Goerli exception:
+            //  * If the network is goerli pre-merge, ignore the extradata check, since we do not
+            //  support clique.
+            if self.chain_spec.chain != Chain::goerli() {
```
Comment (rkrasiuk) [inline crates/consensus/beacon/src/beacon_consensus.rs:71]: how does this differentiate between pre-merge and post-merge goerli?

**Review by rkrasiuk — APPROVED** [review]
lgtm

File: crates/consensus/beacon/src/beacon_consensus.rs (hunk 1 above)
Comment (mattsse) [inline crates/consensus/beacon/src/beacon_consensus.rs:71]: are there any other chains that used clique?

**Review by mattsse — APPROVED** [review]
this check is sufficient.

only checking the chain is reasonable I assume

**Comment by rakita** [comment]
@Rjected how much is block reward in clique, it is probably the problem.

**Comment by Rjected** [comment]
Ah figured it out, we missed a bunch of forks

**Pull request merged**